python/lsst/pipe/tasks/selectImages.py

```diff
@@ -339,3 +339,125 @@ def runDataRef(self, dataRef, coordList, makeDataRefList=True, selectDataList=[]
+            List of ICRS sky coordinates specifying boundary of patch
+        selectDataList : `list` of `SelectStruct`
+            List of SelectStruct, to consider for selection
+        makeDataRefList : `boolean`, optional
+            Construct a list of data references? Default `True`.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Result struct with components:
+            - ``exposureList``: the selected exposures
+                (`list` of `lsst.pipe.tasks.selectImages.BaseExposureInfo`).
+            - ``dataRefList``: the optional data references corresponding to
+                each element of ``exposureList``
+                (`list` of `lsst.daf.persistence.ButlerDataRef`, or `None`).
+
+        Notes
+        -----
+        We use the "convexHull" method of lsst.sphgeom.ConvexPolygon to define
+        polygons on the celestial sphere, and test the polygon of the
+        patch for overlap with the polygon of the image.
+
+        We use "convexHull" instead of generating a ConvexPolygon
+        directly because the standard for the inputs to ConvexPolygon
+        are pretty high and we don't want to be responsible for reaching them.
+        """
+        if self.config.nImagesMax <= 0:
+            raise RuntimeError(f"nImagesMax must be greater than zero: {self.config.nImagesMax}")
+
+        psfSizes = []
+        dataRefList = []
+        exposureInfoList = []
+
+        patchVertices = [coord.getVector() for coord in coordList]
+        patchPoly = lsst.sphgeom.ConvexPolygon.convexHull(patchVertices)
+
+        for data in selectDataList:
+            dataRef = data.dataRef
+            imageWcs = data.wcs
+            cal = dataRef.get("calexp", immediate=True)
+            psfSize = cal.getPsf().computeShape().getDeterminantRadius()
+            nx, ny = cal.getDimensions()
+
+            imageBox = afwGeom.Box2D(afwGeom.Point2D(0, 0), afwGeom.Extent2D(nx, ny))
+            try:
+                imageCorners = [imageWcs.pixelToSky(pix) for pix in imageBox.getCorners()]
+            except (pexExceptions.DomainError, pexExceptions.RuntimeError) as e:
+                # Protecting ourselves from awful Wcs solutions in input images
+                self.log.debug("WCS error in testing calexp %s (%s): deselecting", dataRef.dataId, e)
+                continue
+
+            imagePoly = lsst.sphgeom.ConvexPolygon.convexHull([coord.getVector() for coord in imageCorners])
+            if imagePoly is None:
+                self.log.debug("Unable to create polygon from image %s: deselecting", dataRef.dataId)
+                continue
+
+            if patchPoly.intersects(imagePoly):
+                # "intersects" also covers "contains" or "is contained by",
+                # so there is no guarantee the whole area is covered
+
+                # if min/max PSF values are defined, remove images out of bounds
+                sizeFwhm = psfSize * np.sqrt(8.*np.log(2.))
```

Comment:
Please move the definition of `psfSize` to here, since there's no need for it way up at the start of the method and it's confusing to have it so separated from its usage.